tests/Microsoft.Bot.Builder.Tests/BotStateSetTests.cs

```diff
@@ -88,6 +88,82 @@ public async Task BotStateSet_DualReadWrite()
+            var bs = new BotStateSet()
+                .Use(userState)
+                .Use(convState);
+            var adapter = new TestAdapter()
+                .Use(bs);
+
+            BotCallbackHandler botLogic = async (context, cancellationToken) =>
+            {
+                // get userCount and convCount from botStateSet
+                var userCount = await userProperty.GetAsync(context, () => 100).ConfigureAwait(false);
+                var convCount = await convProperty.GetAsync(context, () => 10).ConfigureAwait(false);
+
+                // System.Diagnostics.Debug.WriteLine($"{context.Activity.Id} UserCount({context.Activity.From.Id}):{userCount} convCount({context.Activity.Conversation.Id}):{convCount}");
+
+                if (context.Activity.Type == ActivityTypes.Message)
+                {
+                    if (context.Activity.Text == "get userCount")
+                    {
+                        await context.SendActivityAsync(context.Activity.CreateReply($"{userCount}"));
+                    }
+                    else if (context.Activity.Text == "get convCount")
+                    {
+                        await context.SendActivityAsync(context.Activity.CreateReply($"{convCount}"));
+                    }
+                }
+
+                // increment userCount and save (since it's a value type, or changed object you don't need to call Set)
+                userCount++;
+                await userProperty.SetAsync(context, userCount);
+
+                // increment convCount and save (since it's a value type, or changed object you don't need to call Set)
+                convCount++;
+                await convProperty.SetAsync(context, convCount);
+            };
+
+            await new TestFlow(adapter, botLogic)
+                .Send("test1")
+                .Send("get userCount")
+                    .AssertReply("101")
+                .Send("get userCount")
+                    .AssertReply("102")
+                .Send("get convCount")
+                    .AssertReply("13")
+                .StartTestAsync();
+
+            // new adapter on new conversation
+            adapter = new TestAdapter(new ConversationReference
+            {
+                ChannelId = "test",
+                ServiceUrl = "https://test.com",
+                User = new ChannelAccount("user1", "User1"),
+                Bot = new ChannelAccount("bot", "Bot"),
+                Conversation = new ConversationAccount(false, "convo2", "Conversation2")
+            })
+                .Use(new BotStateSet(userState, convState));
+
+            await new TestFlow(adapter, botLogic)
+                .Send("get userCount")
+                    .AssertReply("104", "user count should continue on new conversation")
+                .Send("get convCount")
+                    .AssertReply("11", "conversationCount for conversation2 should be reset")
```

Comment:
Consider a constant for "Initial_Conversation_value = 10", and then compare against: 
AssertReply ((INITIAL_VALUE + 1).ToString(); 

That would read more clearly to me that "11" which seems like a magic number.